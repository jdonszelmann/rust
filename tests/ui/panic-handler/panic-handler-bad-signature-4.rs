//@ compile-flags:-C panic=abort

#![no_std]
#![no_main]

use core::panic::PanicInfo;

#[panic_handler]
fn panic<T>(pi: &PanicInfo) -> ! {
    //~^ ERROR #[panic_handler] cannot have generic parameters other than lifetimes
    loop {}
}
