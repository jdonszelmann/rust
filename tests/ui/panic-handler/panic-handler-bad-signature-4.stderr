error: #[panic_handler] cannot have generic parameters other than lifetimes
  --> $DIR/panic-handler-bad-signature-4.rs:9:1
   |
LL | #[panic_handler]
   | ---------------- required by this attribute
LL | fn panic<T>(pi: &PanicInfo) -> ! {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

